test/e2e/localstorage_test.go

```diff
@@ -139,6 +139,135 @@ func TestLocalStorageOperator(t *testing.T) {
 	}
 }
 
+func TestTolerationsLocalStorageOperator(t *testing.T) {
+	localVolumeList := &localv1.LocalVolumeList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LocalVolume",
+			APIVersion: localv1.SchemeGroupVersion.String(),
+		},
+	}
+	err := framework.AddToFrameworkScheme(addToScheme, localVolumeList)
+	if err != nil {
+		t.Fatalf("error adding local volume list : %v", err)
+	}
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = waitForOperatorToBeReady(t, ctx)
+	if err != nil {
+		t.Fatalf("error waiting for operator to be ready : %v", err)
+	}
+
+	f := framework.Global
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatalf("error fetching namespace : %v", err)
+	}
+
+	selectedNode := selectNode(t, f.KubeClient)
+	nodeTaints := selectedNode.Spec.Taints
+	selectedNode.Spec.Taints = []v1.Taint {
+		{
+			Key:	"localstorage",
+			Value:	"testvalue",
+			Effect: "NoSchedule",
+		},
+	}
+	updatedNode, err := waitForNodeTaintUpdate(t, f.KubeClient, selectedNode, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("error tainting node : %v", err)
+	}
+	selectedNode = updatedNode
+
+	selectedDisk, _ := selectDisk(f.KubeClient, selectedNode)
+	waitForPVCreation := true
+	if selectedDisk == "" {
+		waitForPVCreation = false
+		selectedDisk = "/dev/foobar"
+	}
+	localVolume := &localv1.LocalVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LocalVolume",
+			APIVersion: localv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-local-disk",
+			Namespace: namespace,
+		},
+		Spec: localv1.LocalVolumeSpec{
+			Tolerations: []v1.Toleration{
+				{
+					Key:	"localstorage",
+					Value:	"testvalue",
+					Operator: "Equal",
+				},
+			},
+			NodeSelector: &v1.NodeSelector{
```

Comment:
Do we need this node selector?

Reply:
I tested without the NodeSelector, and it will attempt to use all of the nodes. With NodeSelector it takes only a single node (as in the other e2e test), and then will apply/remove the taint during the test.

If the Toleration isn't applied in the LocalVolume, then it will find a node, apply the taint, and be unable to create the DS/Pod as no Node matches the criteria.